exp/api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -102,6 +102,10 @@ type AzureManagedMachinePoolSpec struct {
 	// EnableNodePublicIP controls whether or not nodes in the pool each have a public IP address.
 	// +optional
 	EnableNodePublicIP *bool `json:"enableNodePublicIP,omitempty"`
+
+	// SubnetName selects the Subnet where the MachinePool will be placed
+	// +optional
+	SubnetName string `json:"subnetName,omitempty"`
```

Comment:
shouldn't these subnets also get created as part of the vnet? How does this work when the user is not using a BYO vnet?

Currently the cluster subnet gets created/deleted in the AzureControlPlane controller, additional subnets would need to be added to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/scope/managedcontrolplane.go#L260

Reply:
So the one doubt I have here is, if capz creates a vnet and a subnet and a user creates an additional subnet in the vnet, during the deletion of the cluster do we delete the subnet created by the user as well ?

Reply:
@CecileRobertMichon https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3051
If this issue is resolved, then capz will not need to manage a case when a user is not using a BYO vnet.